src/api/libcellml/component.h

```diff
@@ -49,7 +48,109 @@ class LIBCELLML_EXPORT Component: public Nameable
      */
     void addComponent(const Component &c);
 
+    /**
+     * @brief Remove the component at the given index.
+     * Remove the component from the model with the index @p index.  If the index is
+     * not in the range [0, #components) a std::out_of_range exception
+     * will be thrown.
+     * @param index The index of the component to remove (zero-based).
+     */
+    void removeComponent(size_t index);
+
+    /**
+     * @brief Remove the component with the given name.
+     * Remove the first component found that matches the name @p name.  If the
+     * name is not found no component is removed.
+     *
+     * This is an overloaded method.
+     * @param name The name of the component to remove.
+     */
+    void removeComponent(const std::string &name);
+
+    /**
+     * @brief Tests to see if the component is contained within the Component.
+     * Tests to see if the component with the given name 'name' is contained
+     * within the Component.  Returns true if the component is in the Component and
+     * false otherwise.
+     * @param name The name of the Component to test for existence in the Component.
+     * @return @c true if the named Component is in the model and @c false otherwise.
+     */
+    bool containsComponent(const std::string &name);
+
+    /**
+     * @brief Get a component at index.
+     * Returns a const reference to a component at the index @p index.  If the
+     * index is not valid a std::out_of_range exception is thrown.
+     *
+     * This is an overloaded method.
+     * @param index The index of the Component to return (zero-based).
+     * @return A const reference to the Component at the given index.
+     */
+    const Component& getComponent(size_t index) const;
+
+    /**
+     * @brief Get a component at index.
+     * Returns a reference to a component at the index @p index.  If the index
+     * is not valid a std::out_of_range exception is thrown.
+     *
+     * This is an overloaded method.
+     * @param index The index of the Component to return (zero-based).
+     * @return A reference to the Component at the given index.
+     */
+    Component& getComponent(size_t index);
```

Comment:
Shouldn't we also have ```getComponent()``` functions that take a ```const std::string &name``` as a parameter (i.e. something similar to what has been done for ```removeComponent()```, ```takeComponent()``` and ```replaceComponent()```)?